includes/rest-api/class-amp-rest-fonts-controller.php

```diff
@@ -0,0 +1,256 @@
+<?php
+/**
+ * Class AMP_REST_Fonts_Controller
+ *
+ * @package AMP
+ */
+
+/**
+ * Basic font api for the AMP stories editor.
+ *
+ * Class AMP_REST_Fonts_Controller
+ */
+class AMP_REST_Fonts_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'amp/v1';
+		$this->rest_base = 'fonts';
+	}
+
+	/**
+	 * Registers routes for amp fonts.
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			[
+				[
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => [ $this, 'get_items' ],
+					'permission_callback' => [ $this, 'get_items_permissions_check' ],
+					'args'                => $this->get_collection_params(),
+				],
+				'schema' => [ $this, 'get_public_item_schema' ],
+			]
+		);
+	}
+
+	/**
+	 * Gets a collection of fonts.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 *
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
+		$fonts       = AMP_Fonts::get_fonts();
+		$total_posts = count( $fonts );
+		$page        = $request['page'];
+		$per_page    = $request['per_page'];
+		$max_pages   = ceil( $total_posts / (int) $per_page );
+
+		if ( $page > $max_pages && $total_posts > 0 ) {
+			return new WP_Error( 'rest_post_invalid_page_number', __( 'The page number requested is larger than the number of pages available.', 'amp' ), [ 'status' => 400 ] );
+		}
+
+		$fonts = array_slice( $fonts, ( ( $page - 1 ) * $per_page ), $per_page );
+
+		$formatted_fonts = [];
+		foreach ( $fonts as $font ) {
+			if ( ! $this->check_read_permission( $font ) ) {
+				continue;
+			}
+			$data              = $this->prepare_item_for_response( $font, $request );
+			$formatted_fonts[] = $this->prepare_response_for_collection( $data );
+		}
+
+		$response = rest_ensure_response( $formatted_fonts );
+
+		$response->header( 'X-WP-Total', (int) $total_posts );
+		$response->header( 'X-WP-TotalPages', (int) $max_pages );
+
+		$request_params = $request->get_query_params();
+		$base           = add_query_arg( urlencode_deep( $request_params ), rest_url( sprintf( '%s/%s', $this->namespace, $this->rest_base ) ) );
+
+		if ( $page > 1 ) {
+			$prev_page = $page - 1;
+
+			if ( $prev_page > $max_pages ) {
+				$prev_page = $max_pages;
+			}
+
+			$prev_link = add_query_arg( 'page', $prev_page, $base );
+			$response->link_header( 'prev', $prev_link );
+		}
+		if ( $max_pages > $page ) {
+			$next_page = $page + 1;
+			$next_link = add_query_arg( 'page', $next_page, $base );
+
+			$response->link_header( 'next', $next_link );
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Prepares a single font output for response.
+	 *
+	 * @param Object          $font Comment object.
```

Comment:
```suggestion
	 * @param Object          $font Font object.
```

Reply:
Fixed in https://github.com/ampproject/amp-wp/pull/3998/commits/060b83b2257028b4e06c37bec91b905513ecbb6f